waydroid_extras.py

```diff
@@ -369,104 +315,109 @@ def install_houdini():
+    # sys_overlay_dir = "/var/lib/waydroid/overlay_rw/system"
+    sys_overlay_dir = "/var/lib/waydroid/overlay"
+    merge_dir = "/var/lib/waydroid/rootfs"
+    magisk_dir = os.path.join(sys_overlay_dir, "system", "etc", "init", "magisk")
+    sys_overlay_rw = "/var/lib/waydroid/overlay_rw"
+
+    machine = host()
     init_rc_component = """
 on post-fs-data
     start logd
-    exec - root root -- /system/etc/init/magisk/magisk{arch} --setup-sbin /system/etc/init/magisk
-    exec - root root -- /system/etc/init/magisk/magiskpolicy --live --magisk "allow * magisk_file lnk_file *"
+    exec u:r:su:s0 root root -- /system/etc/init/magisk/magisk{arch} --auto-selinux --setup-sbin /system/etc/init/magisk
+    exec u:r:su:s0 root root -- /system/etc/init/magisk/magiskpolicy --live --magisk "allow * magisk_file lnk_file *"
     mkdir /sbin/.magisk 700
     mkdir /sbin/.magisk/mirror 700
     mkdir /sbin/.magisk/block 700
     copy /system/etc/init/magisk/config /sbin/.magisk/config
     rm /dev/.magisk_unblock
-    start FAhW7H9G5sf
+    start 7zKkuZ1ZhD
     wait /dev/.magisk_unblock 40
     rm /dev/.magisk_unblock
-service FAhW7H9G5sf /sbin/magisk --post-fs-data
+
+service 7zKkuZ1ZhD /sbin/magisk --auto-selinux --post-fs-data
     user root
-    seclabel -
+    seclabel u:r:su:s0
     oneshot
-service HLiFsR1HtIXVN6 /sbin/magisk --service
+
+service wHgGlkRCtMoIQw /sbin/magisk --auto-selinux --service
     class late_start
     user root
-    seclabel -
+    seclabel u:r:su:s0
     oneshot
+
 on property:sys.boot_completed=1
     mkdir /data/adb/magisk 755
-    exec - root root -- /sbin/magisk --boot-complete
-    exec -- /system/bin/sh -c "if [ ! -e /data/data/io.github.huskydg.magisk ] ; then pm install /system/etc/init/magisk/magisk.apk; fi"
+    exec u:r:su:s0 root root -- /sbin/magisk --auto-selinux --boot-complete
+    exec -- /system/bin/sh -c "if [ ! -e /data/data/io.github.huskydg.magisk ] ; then pm install /system/etc/init/magisk/magisk.apk ; fi"
    
 on property:init.svc.zygote=restarting
-    exec - root root -- /sbin/magisk --zygote-restart
+    exec u:r:su:s0 root root -- /sbin/magisk --auto-selinux --zygote-restart
    
 on property:init.svc.zygote=stopped
-    exec - root root -- /sbin/magisk --zygote-restart
-    """.format(arch=32 if arch=="" else 64)
-    
-    system_img = os.path.join(get_image_dir(), "system.img")
-    if not os.path.isfile(system_img):
-        print("The system image path '{}' from waydroid config is not valid !".format(system_img))
-        sys.exit(1)
-    print("==> Found system image: " + system_img)
-
-    img_size = int(os.path.getsize(system_img)/(1024*1024))
+    exec u:r:su:s0 root root -- /sbin/magisk --auto-selinux --zygote-restart
+    """.format(arch=machine[1])
 
-    # Resize image to get some free space
-    resize_img(system_img, "{}M".format(img_size+50))
+    if not DBusContainerService().GetSession():
+        print("Please make sure waydroid container is running!")
```

Comment:
I should note that it is possible to opt-out of D-Bus service activation entirely in Waydroid 1.4.0, see `USE_DBUS_ACTIVATION` variable in https://github.com/waydroid/waydroid/blob/main/Makefile; from what I can tell this check would make it impossible to install Magisk on a Waydroid installation with `USE_DBUS_ACTIVATION=0` set (since there would be no `id.waydro.Container.service`)

Reply:
Here I think it can be changed to check whether `/var/lib/waydroid/rootfs/system/etc/init/bootanim.rc` exists